From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 19 Nov 2025 10:12:50 -0600
Subject: pmdomain: rockchip: quiet regulator error on -EPROBE_DEFER

Change the dev_err() to dev_err_probe() under rockchip_pd_power_on()
to prevent errors early in the boot process when the requested
regulator is not yet available. This converts errors like the following
to debug messages:

rockchip-pm-domain fd8d8000.power-management:power-controller: Failed to enable supply: -517

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 drivers/pmdomain/rockchip/pm-domains.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pmdomain/rockchip/pm-domains.c b/drivers/pmdomain/rockchip/pm-domains.c
index 111111111111..222222222222 100644
--- a/drivers/pmdomain/rockchip/pm-domains.c
+++ b/drivers/pmdomain/rockchip/pm-domains.c
@@ -706,7 +706,8 @@ static int rockchip_pd_power_on(struct generic_pm_domain *domain)
 
 	ret = rockchip_pd_regulator_enable(pd);
 	if (ret) {
-		dev_err(pd->pmu->dev, "Failed to enable supply: %d\n", ret);
+		dev_err_probe(pd->pmu->dev, ret,
+			      "Failed to enable supply: %d\n", ret);
 		return ret;
 	}
 
-- 
Armbian

